src/univariate/continuous/normal.jl

```diff
@@ -211,16 +211,16 @@ end
 
 # fit_mle based on sufficient statistics
 
-fit_mle(::Type{<:Normal}, ss::NormalStats) = Normal(ss.m, sqrt(ss.s2 / ss.tw))
+fit_mle(::Type{D}, ss::NormalStats) where {D<:Normal} = D(ss.m, sqrt(ss.s2 / ss.tw))
 fit_mle(g::NormalKnownMu, ss::NormalKnownMuStats) = Normal(g.μ, sqrt(ss.s2 / ss.tw))
 fit_mle(g::NormalKnownSigma, ss::NormalKnownSigmaStats) = Normal(ss.sx / ss.tw, g.σ)
 
 # generic fit_mle methods
 
-function fit_mle(::Type{<:Normal}, x::AbstractArray{T}; mu::Float64=NaN, sigma::Float64=NaN) where T<:Real
+function fit_mle(::Type{D}, x::AbstractArray{<:Real}; mu::Float64=NaN, sigma::Float64=NaN) where {D<:Normal}
```

Comment:
The keyword arguments should be relaxed as well. I suggest changing their default values to `nothing`, as this allows a type stable implementation (similar to e.g. `mean` in `var`: https://docs.julialang.org/en/v1/stdlib/Statistics/#Statistics.var) and avoids promotion of e.g. `Float32` data:
```suggestion
function fit_mle(::Type{D}, x::AbstractArray{<:Real}; mu::Union{Nothing,Real}=nothing, sigma::Union{Nothing,Real}=nothing) where {D<:Normal}
```